packages/core/src/generators/init/templates/root/Directory.Build.targets__tmpl__

```diff
@@ -0,0 +1,13 @@
+<!-- 
+  This file is imported late in the build order. 
+  Use it to override properties and define dependent properties.
+-->
+<Project>
+  <PropertyGroup>
+    <NxProjectName>$([System.IO.Path]::GetFileName($(MSBuildProjectDirectory)))</NxProjectName>
+    <NodeModulesRelativePath>$([System.IO.Path]::GetRelativePath($(MSBuildProjectDirectory), $(RepoRoot)))</NodeModulesRelativePath>
+  </PropertyGroup>
+  <Target Name="CheckNxModuleBoundaries" BeforeTargets="Build">    
+    <Exec Command="node $(NodeModulesRelativePath)/<%= checkModuleBoundariesScriptPath %> -p $(NxProjectName)"/>
+  </Target>
```

Comment:
I'm not sure this would be accurate when dealing with projects that's names don't match the file structure. Since we control the node script too, let's update it to take the project root as a parameter. Then we could look for the project based on its root path.

Reply:
We should add project-root as a parameter and accept either... https://github.com/nx-dotnet/nx-dotnet/blob/master/packages/core/src/tasks/check-module-boundaries.ts#L90-L95

If project provided use it, if project-root provided look up the project.

Reply:
From my understanding of `getProjectFileForNxProject`, there should only be one dotnet project under each nx project root. We could overload the `project` argument, so if file path value with file extension is provided (csproj, vbproj, etc.), then it looks for the nx project configuration where the root is part of the path. Alternative would be to have a different argument if there's a good name for it.

Reply:
I'd prefer to use a different argument, s.t. the original parameter doesn't have any new modality added to it. I'd use `--project-root` as the flag, should be added as `projectRoot` in the lines I linked above.